### PR TITLE
update storage README

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/README.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+Learn more about options for authentication (including Connection Strings, Shared Key, Shared Key Signatures, Active Directory, and anonymous public access) in our [samples](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/storage/azure-storage-blobs/samples).
+
+### Thread safety
+
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/cpp_introduction.html#thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+### Additional concepts
+
+TBD
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/README.md:79]: Hi @RickWinter , I'm looking at README for blob package in .Net SDK. They also introduced some additional concepts of client options, LROs, error handlings and how to access raw responses etc., which could be super helpful for some customers according to the feedback/questions we've got on GitHub or via other channels.

As those are more general concepts that are not specific to any service SDK, introduction of those concepts are linked to Azure Core sdk in .Net. Do you think we can do the same in C++ SDK?

**Review by microzchang — APPROVED** [review]

**Comment by antkmsft** [comment]
@Jinming-Hu, this PR is a little old, do you still need it? Did you get the info you needed from Rick?

**Comment by Jinming-Hu** [comment]
@antkmsft I'll resume this work when I finish the test-proxy thing.

File: sdk/storage/README.md [hunk 2]
```diff
@@ -1,139 +1,35 @@
-# Azure Storage Client Library for C++
+# Azure Storage libraries for C++
```
Comment (ahsonkhan) [inline sdk/storage/README.md:1]: FYI, this change broke the anchor link in the root README file that references it:
https://github.com/Azure/azure-sdk-for-cpp#getting-started

Comment (ahsonkhan) [reply sdk/storage/README.md:1]: Fixed in https://github.com/Azure/azure-sdk-for-cpp/pull/4788

**Pull request merged**